rust/cardano-blockchain-types/src/point.rs

```diff
@@ -0,0 +1,595 @@
+    #[must_use]
+    pub fn fuzzy(slot: Slot) -> Self {
+        Self(pallas::network::miniprotocols::Point::Specific(
+            slot.into(),
+            Vec::new(),
+        ))
+    }
+
+    /// Creates a new Fuzzy `Point` from a concrete point.
+    ///
+    /// Will not alter either TIP or ORIGIN points.
+    #[must_use]
+    pub fn as_fuzzy(&self) -> Self {
+        if *self == Self::TIP {
+            Self::TIP
+        } else {
+            match self.0 {
+                pallas::network::miniprotocols::Point::Specific(slot, _) => {
+                    Self::fuzzy(slot.into())
+                },
+                pallas::network::miniprotocols::Point::Origin => Self::ORIGIN,
+            }
+        }
+    }
+
+    /// Check if a Point is Fuzzy.
+    ///
+    /// Even though we don't know the hash for TIP or Origin, neither of these points
+    /// are considered to be fuzzy.
+    ///
+    /// # Examples
+    ///
+    /// ```rs
+    /// use cardano_chain_follower::Point;
+    ///
+    /// let slot = 42;
+    /// let point = Point::fuzzy(slot);
+    ///
+    /// assert!(point.is_fuzzy());
+    /// ```
+    #[must_use]
+    pub fn is_fuzzy(&self) -> bool {
+        if *self == Self::TIP {
+            false
+        } else {
+            match self.0 {
+                pallas::network::miniprotocols::Point::Specific(_, ref hash) => hash.is_empty(),
+                pallas::network::miniprotocols::Point::Origin => false,
+            }
+        }
+    }
+
+    /// Check if a Point is the origin.
+    ///
+    /// Origin is the synthetic Origin point, and ALSO any point thats at slot zero with a
+    /// hash.
+    ///
+    /// # Examples
+    ///
+    /// ```rs
+    /// use cardano_chain_follower::Point;
+    ///
+    /// let slot = 42;
+    /// let point = Point::fuzzy(slot);
+    ///
+    /// assert!(!point.is_origin());
+    /// ```
+    #[must_use]
+    pub fn is_origin(&self) -> bool {
+        match self.0 {
+            pallas::network::miniprotocols::Point::Specific(slot, ref hash) => {
+                slot == 0 && !hash.is_empty()
+            },
+            pallas::network::miniprotocols::Point::Origin => true,
+        }
+    }
+
+    /// Check if a Point is actually unknown.
+    ///
+    /// # Examples
+    ///
+    /// ```rs
+    /// use cardano_chain_follower::Point;
+    ///
+    /// let point = Point::fuzzy(0);
+    ///
+    /// assert!(point.is_unknown());
+    /// ```
+    #[must_use]
+    pub fn is_unknown(&self) -> bool {
+        match self.0 {
+            pallas::network::miniprotocols::Point::Specific(slot, ref hash) => {
+                slot == 0 && hash.is_empty()
+            },
+            pallas::network::miniprotocols::Point::Origin => false,
+        }
+    }
+
+    /// Check if a Point is actually unknown.
+    ///
+    /// # Examples
+    ///
+    /// ```rs
+    /// use cardano_chain_follower::Point;
+    ///
+    /// let point = Point::fuzzy(0);
+    ///
+    /// assert!(point.is_unknown());
```

Comment:
```suggestion
    /// Check if a Point is the tip.
    ///
    /// # Examples
    ///
    /// ```rs
    /// use cardano_chain_follower::Point;
    ///
    /// let point = Point::fuzzy(0);
    ///
    /// assert!(!point.is_tip());
```